charts/kcp/templates/etcd-statefulset.yaml

```
@@ -114,4 +114,7 @@ spec:
         resources:
           requests:
             storage: {{ .Values.etcd.volumeSize }}
+        {{- if .Values.etcd.storageClassName }}
+        storageClassName: {{ .Values.etcd.storageClassName }}
+        {{- end }}
```

small nit to clean it up

```suggestion
        {{- with .Values.etcd.storageClassName }}
        storageClassName: {{ . }}
        {{- end }}
```

---

With the last commit, we have:
```yaml
            - name: client-ca
              mountPath: /etc/etcd/tls/client-ca
      volumes:
        - name: peer-cert
          secret:
            secretName: kcp-etcd-peer-cert
        - name: peer-ca
          secret:
            secretName: kcp-etcd-peer-ca
        - name: server-cert
          secret:
            secretName: kcp-etcd-cert
        - name: client-ca
          secret:
            secretName: kcp-etcd-client-ca
  volumeClaimTemplates:
    - metadata:
        name: etcd-data
      spec:
        accessModes: ["ReadWriteOnce"]
        resources:
          requests:
            storage: 8Gi
```
when not overridden. And:
```yaml
            secretName: kcp-etcd-client-ca
  volumeClaimTemplates:
    - metadata:
        name: etcd-data
      spec:
        accessModes: ["ReadWriteOnce"]
        resources:
          requests:
            storage: 8Gi
        storageClassName: premium-rwo
```
when overridden.